### PR TITLE
cli: minor updates to `jj util completion`

### PR DESCRIPTION
**Pull request by ilyagr**
A few minor updates after 0f27152 AKA #2945

Specifically, I tried to change the help text so that it looks better as Markdown. I also reworded the deprecation warning and added a hint.

### CODE REVIEW

**Review by PhilipMetzger — APPROVED** [review]

File: cli/src/commands/util.rs [hunk 1]
```diff
@@ -34,15 +34,21 @@ pub(crate) enum UtilCommand {
     ConfigSchema(UtilConfigSchemaArgs),
 }
 
+// TODO: rustfmt is disabled for the entire struct. Can we make it affect the
+// docstring only?
```
Comment (jyn514) [inline cli/src/commands/util.rs:38]: it's kinda hacky, but this works:
```diff
diff --git a/cli/src/commands/util.rs b/cli/src/commands/util.rs
index 69f3f914..172c8551 100644
--- a/cli/src/commands/util.rs
+++ b/cli/src/commands/util.rs
@@ -44,11 +44,11 @@ pub(crate) enum UtilCommand {
 /// - **bash**: `source <(jj util completion)`
 /// - **fish**: `jj util completion --fish | source`
 /// - **zsh**:
-///     ```shell
-///     autoload -U compinit
-///     compinit
-///     source <(jj util completion --zsh)
-///     ```
+#[doc = "     ```shell
+     autoload -U compinit
+     compinit
+     source <(jj util completion --zsh)
+     ```"]
 #[derive(clap::Args, Clone, Debug)]
 #[command(verbatim_doc_comment)]
 pub(crate) struct UtilCompletionArgs {

```

**Review by jyn514 — APPROVED** [review]
this is great, thanks!

File: cli/src/commands/util.rs (hunk 1 above)
Comment (ilyagr) [reply cli/src/commands/util.rs:38]: I suppose we could specify the entire docstring as an explicit `doc` attribute. I'm going back and forth on whether I like the idea.

Comment (ilyagr) [reply cli/src/commands/util.rs]: I guess it looks OK. :) Thanks!

**Comment by martinvonz** [comment]
I think the `jj util completion --help` output got a bit uglier with this PR. Maybe we'll want to produce the help output ourselves in the long term. Then we can use some library for formatting markdown for the terminal, and we can do other things like grouping commands under headings to make them easier to find.

**Comment by ilyagr** [comment]
If I don't forget, I'll make a PR deleting the `**` around the shells. I think that will look good enough to me.

For custom output, one library that does this is https://github.com/Canop/clap-help?tab=readme-ov-file. It makes some choices I don't like (it doesn't like long option descriptions; more personally, I don't like the color scheme), but it is an example of how that can be done. The library it uses to output markdown is supposed to be quite configurable, though as I mentioned, I don't like the default config.

**Pull request merged**